### PR TITLE
PUBDEV-6646: Create SharedTreeMojoModelWithContributions class to have central location of contribs for DRF and GBM MOJO

### PR DESCRIPTION
**Pull request by navdeep-G**
* Follow up from https://github.com/h2oai/h2o-3/pull/3628

### CODE REVIEW

File: h2o-genmodel/src/main/java/hex/genmodel/algos/tree/SharedTreeMojoModelWithContributions.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        TreeSHAPPredictor<double[]> predictor = new TreeSHAPEnsemble<>(treeSHAPs, (float) getInitF());
+        
+        return getContributionsPredictor(predictor);
+    }
+
+    public double getInitF() {
+        return 0; // Set to zero by default, which is correct for DRF. However, need to override in GBMMojoModel with correct init_f.
+    }
+    
+    protected abstract ContributionsPredictor getContributionsPredictor(TreeSHAPPredictor<double[]> treeSHAPPredictor);
```
Comment (honzasterba) [inline h2o-genmodel/src/main/java/hex/genmodel/algos/tree/SharedTreeMojoModelWithContributions.java]: abstract class with no abstract methods

File: h2o-genmodel/src/main/java/hex/genmodel/algos/drf/DrfMojoModel.java [hunk 2]
```diff
@@ -2,25 +2,26 @@
...
     protected boolean _binomial_double_trees;
 
 
     public DrfMojoModel(String[] columns, String[][] domains, String responseColumn) {
         super(columns, domains, responseColumn);
     }
 
+    @Override
+    protected ContributionsPredictor getContributionsPredictor(TreeSHAPPredictor<double[]> treeSHAPPredictor) {
+        return new ContributionsPredictorDRF(treeSHAPPredictor){
```
Comment (honzasterba) [inline h2o-genmodel/src/main/java/hex/genmodel/algos/drf/DrfMojoModel.java]: this class is not abstract yet you have an annonymous inner class here

File: h2o-genmodel/src/main/java/hex/genmodel/algos/tree/SharedTreeMojoModelWithContributions.java (hunk 1 above)
Comment (navdeep-G) [reply h2o-genmodel/src/main/java/hex/genmodel/algos/tree/SharedTreeMojoModelWithContributions.java]: Good point. Done.

File: h2o-genmodel/src/main/java/hex/genmodel/algos/drf/DrfMojoModel.java (hunk 2 above)
Comment (navdeep-G) [reply h2o-genmodel/src/main/java/hex/genmodel/algos/drf/DrfMojoModel.java]: Good point. Done.

Comment (honzasterba) [inline h2o-genmodel/src/main/java/hex/genmodel/algos/drf/DrfMojoModel.java]: unnecessary {}

File: h2o-genmodel/src/main/java/hex/genmodel/algos/gbm/GbmMojoModel.java [hunk 3]
```diff
@@ -24,6 +19,16 @@ public GbmMojoModel(String[] columns, String[][] domains, String responseColumn)
         super(columns, domains, responseColumn);
     }
 
+    @Override
+    protected ContributionsPredictor getContributionsPredictor(TreeSHAPPredictor<double[]> treeSHAPPredictor) {
+        return new ContributionsPredictor(treeSHAPPredictor){
```
Comment (honzasterba) [inline h2o-genmodel/src/main/java/hex/genmodel/algos/gbm/GbmMojoModel.java]: unnecessary {}

File: h2o-genmodel/src/main/java/hex/genmodel/algos/drf/DrfMojoModel.java (hunk 2 above)
Comment (navdeep-G) [reply h2o-genmodel/src/main/java/hex/genmodel/algos/drf/DrfMojoModel.java]: Done.

File: h2o-genmodel/src/main/java/hex/genmodel/algos/gbm/GbmMojoModel.java (hunk 3 above)
Comment (navdeep-G) [reply h2o-genmodel/src/main/java/hex/genmodel/algos/gbm/GbmMojoModel.java]: Done.

**Comment by navdeep-G** [comment]
@honzasterba There is no JIRA for this. Should there be?

**Comment by navdeep-G** [comment]
I have added a link to JIRA in the PR title @honzasterba

**Pull request merged**